common/server.cpp

```diff
@@ -168,6 +170,15 @@ AuthenticationResult Server::loginUser(Server_ProtocolHandler *session, QString
     event.mutable_user_info()->CopyFrom(session->copyUserInfo(true, true, true));
     locker.unlock();
     
+    // check if client id exists (older client compatibility)
+    if (clientid.isEmpty()){
+        // client id is empty, either out dated client or client has been modified
```

Comment:
I started to add the functionality for requiring the client ID (as you can see in the response protocol place holder) but decided to hold off till we decided what was to be done with the client IDs.